### PR TITLE
tests: log if no samples

### PR DESCRIPTION
**Pull request by acpana**
While running the samples tests it may be helpful to know if no samples load and exist early.

### CODE REVIEW

File: config/tests/samples/create/samples_test.go [hunk 1]
```diff
@@ -233,6 +233,11 @@ func TestAll(t *testing.T) {
 
 	setup()
 	samples := LoadMatchingSamples(t, regexp.MustCompile(runTestsRegex), project)
+	if len(samples) == 0 {
+		t.Logf("No tests to run for pattern %s", runTestsRegex)
```
Comment (justinsb) [inline config/tests/samples/create/samples_test.go:237]: Nit: Does this happen?  Should this be t.Errorf?   It's an odd edge case ... t.Skipf might be the most appropriate!  (This is why we use nested tests elsewhere!)

Comment (acpana) [reply config/tests/samples/create/samples_test.go:237]: Actually I quite like erroring out as it will be a better signal that something is wrong.

Yeah, with the tests as is, it happened to me when I didn't figure out how to use `run-tests` correctly 😅 . So:

```
# misspel
$ go test -v -tags=integration ./config/tests/samples/create -timeout 1600s -test.count=1 -test.run TestAll -run-tests gfehub 2>&1

# not a thing / i'm holding it wrong
$ go test -v -tags=integration ./config/tests/samples/create -timeout 1600s -test.count=1 -test.run TestAll -run-tests notathing 2>&1
```

come out as 

```
...
ok      github.com/GoogleCloudPlatform/k8s-config-connector/config/tests/samples/create 11.430s
```

which can offer a false sense of security hehe.

Comment (maqiuyujoyce) [reply config/tests/samples/create/samples_test.go:237]: I think error out gives a more clear signal. We can always make improvements in the future.
